File: RestaurantReservationAPI/Models/Availability.cs
namespace RestaurantReservationAPI.Models;

public class Availability
{
  public DateTime ReservationTime { get; set; }
  public int TableCount { get; set; }
}

public class AvailabilityForDay
{
  public required int Day { get; set; }
  public required int State { get; set; }
}

File: restaurantclient/src/api/models/availability.ts
export interface AvailabilityByMonth {
  day: number;
  state: number;
}

export interface AvailabilityByDay {
  reservationTime: Date;
  tableCount: number;
}


const stateMapping: { [key: number]: string } = {
  0: 'inPast',
  1: 'today',
  2: 'free',
  3: 'allBooked',
  4: 'closed',
};
export function getState(state: number): string {
  return stateMapping[state] || 'unknown';
}
